### PR TITLE
Add dcd_edpt_close() to esp32-s2

### PR DESCRIPTION
**Pull request by Milili-Tech**
**Describe the PR**
Implement dcd_edpt_close() to esp32-s2. This referred to st/synopsys implementation.

**Additional context**
The examples which support esp32-s2 are tested, include a modification version of uac2_headset.

Relate to #340.

### CODE REVIEW

**Comment by hathach** [comment]
thank you for your PR, can I ask whether you work for espressif. Since you seem to undo their PR https://github.com/hathach/tinyusb/pull/454 

esp32s2 has less fifo than endpoint, which is `_allocated_fifos` for, it is the number of fifos.

**Comment by JeffGant** [comment]
@hathach have reverted this part back, please check.

File: src/portable/espressif/esp32s2/dcd_esp32s2.c [hunk 1]
```diff
@@ -127,7 +127,7 @@ static void bus_reset(void)
   //   Recommended value = 10 + 1 + 2 x (16+2) = 47 --> Let's make it 52
   USB0.grstctl |= 0x10 << USB_TXFNUM_S; // fifo 0x10,
   USB0.grstctl |= USB_TXFFLSH_M;        // Flush fifo
-  USB0.grxfsiz = 52;
+  USB0.grxfsiz = 15 + 2 * (256/4) + 2*EP_MAX;
```
Comment (me-no-dev) [inline src/portable/espressif/esp32s2/dcd_esp32s2.c:130]: can you please explain this calculation?

Comment (JeffGant) [reply src/portable/espressif/esp32s2/dcd_esp32s2.c:130]: It's a workaround provided by @hathach in #540 , but actually need better solution.

Comment (hathach) [reply src/portable/espressif/esp32s2/dcd_esp32s2.c:130]: this is based on recommended by stm32 guide, however, it is not optimal for non-ISO application. I plan to work on a better fifo allocation scheme later on. I think it is probably best to leave this PR pending until we get thing settled with stm32 first. 2 moving targets are harder to work with than 1.

**Pull request closed without merging**